feat: add readme and diagrams (#1)
kkrt-labs/kakarot-indexer: README.md

# Kakarot Indexer

> Kakarot Indexer is part of the three-part architecture of the Kakarot zkEVM rollup (Kakarot EVM Cairo Programs, Kakarot RPC, Kakarot Indexer). Its job is to index the underlying CairoVM network on which Kakarot core is deployed in an Ethereum compatible format.

Reviewer: ```suggestion
Kakarot Indexer fits in the three-part architecture of the Kakarot zkEVM rollup (Kakarot EVM Cairo Programs, Kakarot RPC, Kakarot Indexer). It will index the underlying CairoVM network on which Kakarot core is deployed in an Ethereum compatible format.
```

Author: Done
